lib/semigrp.gi

```diff
@@ -542,7 +542,25 @@ InstallMethod( AsSemigroup,
     "generic method for collections",
     [ IsCollection ],
     function ( D )
-    local   S,  L;
+    local L, S, i, j, k;
+
+    # Must check associativity
+    if ApplicableMethod(IsAssociative, [D]) <> fail then
```

Comment:
Perhaps also
```suggestion
    if HasIsAssociative(D) or ApplicableMethod(IsAssociative, [D]) <> fail then
```

Reply:
Is it the case that if something has `HasIsAssociative`, then it necessarily has an applicable method (i.e. the 'getter'?)

Reply:
Indeed, I think you are right. Might be worthwhile adding this as a comment -- but actually I'd greatly prefer if we found a way to avoid the use of `ApplicableMethod` in the first place :-). But that's just taste, I won't object in the end either way.